### PR TITLE
[Gst-va] add vp9 10bit lp encode support

### PR DESCRIPTION
**Pull request by FocusLuo**
add vp9 10bit lp encode support

### CODE REVIEW

File: test/gst-va/encode/10bit/vp9.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+spec = load_test_spec("vp9", "encode", "10bit")
+
+@slash.requires(*have_gst_element("vavp9lpenc"))
+@slash.requires(*have_gst_element("vavp9dec"))
+class VP9_10EncoderBaseTest(EncoderTest):
+  def before(self):
+    super().before()
+    vars(self).update(
+      codec         = "vp9",
+      gstencoder    = "vavp9lpenc",
```
Comment (uartie) [inline test/gst-va/encode/10bit/vp9.py:20]: This should be in the `VP9_10EncoderLPTest` subclass

File: test/gst-va/encode/10bit/vp9.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from .....lib import *
+from .....lib.gstreamer.va.util import *
+from .....lib.gstreamer.va.encoder import EncoderTest
+
+spec = load_test_spec("vp9", "encode", "10bit")
+
+@slash.requires(*have_gst_element("vavp9lpenc"))
```
Comment (uartie) [inline test/gst-va/encode/10bit/vp9.py:13]: This should be at the `VP9_10EncoderLPTest` subclass

File: test/gst-va/encode/10bit/vp9.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      frames    = vars(self).get("brframes", self.frames),
+      gop       = gop,
+      maxrate   = bitrate,
+      minrate   = bitrate,
+      rcmode    = "cbr",
+    )
+
+  @parametrize_with_unused(*gen_vp9_cbr_lp_parameters(spec), ['slices','refmode', 'looplvl', 'loopshp'])
+  def test(self, case, gop, bitrate, fps, slices, refmode, looplvl, loopshp):
+    self.init(spec, case, gop, bitrate, fps)
```
Comment (uartie) [inline test/gst-va/encode/10bit/vp9.py:72]: `looplvl`, `loopshp` should be used.

**Review by uartie — CHANGES_REQUESTED** [review]

File: test/gst-va/encode/10bit/vp9.py (hunk 1 above)
Comment (FocusLuo) [reply test/gst-va/encode/10bit/vp9.py:18]: updated

File: test/gst-va/encode/10bit/vp9.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      case      = case,
+      gop       = 30 if ipmode != 0 else 1,
+      qp        = qp,
+      quality   = quality,
+      rcmode    = "cqp",
+    )
+
+  @parametrize_with_unused(*gen_vp9_cqp_lp_parameters(spec), ['slices','refmode', 'looplvl', 'loopshp'])
+  def test(self, case, ipmode, qp, quality, slices, refmode, looplvl, loopshp):
+    self.init(spec, case, ipmode, qp, quality)
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply test/gst-va/encode/10bit/vp9.py]: updated

File: test/gst-va/encode/10bit/vp9.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      # target percentage 50%
+      maxrate   = bitrate * 2,
+      minrate   = bitrate,
+      quality   = quality,
+      rcmode    = "vbr",
+    )
+
+  @parametrize_with_unused(*gen_vp9_vbr_lp_parameters(spec), ['slices','refmode', 'looplvl', 'loopshp'])
+  def test(self, case, gop, bitrate, fps, quality, slices, refmode, looplvl, loopshp):
+    self.init(spec, case, gop, bitrate, fps, quality)
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply test/gst-va/encode/10bit/vp9.py]: updated

File: test/gst-va/encode/10bit/vp9.py (hunk 3 above)
Comment (FocusLuo) [reply test/gst-va/encode/10bit/vp9.py]: updated

File: test/gst-va/encode/10bit/vp9.py (hunk 2 above)
Comment (FocusLuo) [reply test/gst-va/encode/10bit/vp9.py]: updated

**Comment by FocusLuo** [comment]
Updated, thanks detail review feedback

File: test/gst-va/encode/10bit/vp9.py (hunk 1 above)
Comment (uartie) [reply test/gst-va/encode/10bit/vp9.py:18]: Hmm.. not updated

File: test/gst-va/encode/10bit/vp9.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      gstparser     = "vp9parse",
+      gstmuxer      = "matroskamux",
+      gstdemuxer    = "matroskademux",
+    )
+
+  def get_file_ext(self):
+    return "webm"
+
+@slash.requires(*have_gst_element("vavp9lpenc"))
+@slash.requires(*have_gst_element("vavp9dec"))
```
Comment (uartie) [inline test/gst-va/encode/10bit/vp9.py:30]: This needs to go to the VP9_10EncoderBaseTest

**Review by uartie — CHANGES_REQUESTED** [review]

File: test/gst-va/encode/10bit/vp9.py (hunk 6 above)
Comment (FocusLuo) [reply test/gst-va/encode/10bit/vp9.py]: Updated,
I missed understand last time, how about now?

File: test/gst-va/encode/10bit/vp9.py (hunk 1 above)
Comment (FocusLuo) [reply test/gst-va/encode/10bit/vp9.py]: Updated,
I missed understand last time, how about now?

**Review by uartie — APPROVED** [review]
LGTM

File: test/gst-va/encode/10bit/vp9.py (hunk 6 above)
Comment (uartie) [reply test/gst-va/encode/10bit/vp9.py]: Yep. Thanks!

**Comment by uartie** [comment]
Test failures not caused by this PR.

**Pull request merged**